src/napari_cryoet_data_portal/_reader.py

```diff
@@ -1,22 +1,71 @@
 """Functions to read data from the portal into napari types."""
 
-from typing import Any, Dict, List, Optional, Tuple
+import warnings
+from typing import Any, Dict, Generator, List, Optional, Tuple
 import fsspec
 
+import numpy as np
 import ndjson
 from napari_ome_zarr import napari_get_reader
 from npe2.types import FullLayerData, PathOrPaths, ReaderFunction
-from cryoet_data_portal import Annotation, Tomogram
+from cryoet_data_portal import Annotation, AnnotationFile, Tomogram
+from cmap import Colormap
+from napari.utils.colormaps import direct_colormap
 
 from napari_cryoet_data_portal._logging import logger
 
-
-OBJECT_COLOR = {
-    "ribosome": "red",
-    "ribosome, 80 s": "red",
-    "fatty acid synthase": "darkblue",
+# Maps Annotation.object_name to a color.
+# The object names were manually generated using the following code.
+#
+# from cryoet_data_portal import Annotation, Client
+# client = Client()
+# annotations = Annotation.find(client)
+# names = tuple(a.object_name for a in annotations)
+#
+# TODO: add a tool to update these or do this automatically when
+# starting the plugin.
+OBJECT_NAMES = (
```

Comment:
This list is absolutely going to evolve quickly -- what happens when new object types get added to the portal and they don't match this list?

Reply:
I could run the code snippet at some point (e.g. at import, the first time a color is needed). That will be stable until there is a new object name, which is probably good enough for me. What do you think? Any other ideas?

Reply:
Maybe I could just use `OBJECT_COLORMAP(int(Annotation.object_id) % len(OBJECT_COLORMAP.color_stops.stops)`?

I'm not sure how spread out those will be, but I'm assuming the ontology IDs are stable.

Reply:
They're all in the format 'GO:0001234' so for the forseeable future you could split them on `:` and cast them to ints, I think that would do pretty well for the moment!

Reply:
I did this. Thanks for pointing out the need for the split.